test/libsolidity/syntaxTests/parsing/mapping_with_names_conflict_3.sol

```diff
@@ -0,0 +1,7 @@
+contract test {
+    mapping(address owner => mapping(address owner => address owner)) owner;
+}
+// ----
+// DeclarationError 5609: (45-84): Conflicting parameter name "owner" in mapping.
+// DeclarationError 1809: (20-85): Conflicting parameter name "owner" in recursive mapping.
```

Comment:
Why does this one have a different error code, i.e. 1809 vs 5609?

Reply:
Since both these cases serve the same purpose, it might make sense to just have the same error and code in both cases.

> so you're differentiating between conflicting key -> key and key -> value names

I did that because the error logic was at different places in the code and I tend to write more descriptive error msg if possible. But here I agree the solidity dev that ends up on this error doesn't benefit much.